code-challenges.test.js

```diff
@@ -1,67 +1,326 @@
+       // Ran all test suites.
+       // ✨  Done in 0.63s.
+ 
+   // Refactor, if necessary
+ 
+ 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
-
+ 
 // a) Create a test with expect statements for each of the variables provided.
-
+ 
+ 
 const accountTransactions1 = [100, -17, -23, -9]
 // Expected output: [100, 83, 60, 51]
-
+ 
 const accountTransactions2 = [250, -89, 100, -96]
 // Expected output: [250, 161, 261, 165]
-
+ 
 const accountTransactions3 = []
 // Expected output: []
-
+ 
+// RED
+ // Write the test
+     describe("accumulatingSum", () => {
+       it("takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.", () => {
+         const accountTransactions1 = [100, -17, -23, -9]
+         // Expected output: [100, 83, 60, 51]
+ 
+         const accountTransactions2 = [250, -89, 100, -96]
+         // Expected output: [250, 161, 261, 165]
+ 
+         const accountTransactions3 = []
+         // Expected output: [] 
+         expect(accumulatingSum(accountTransactions1)).toEqual([100, 83, 60, 51])
+         expect(accumulatingSum(accountTransactions2)).toEqual([250, 161, 261, 165])
+         expect(accumulatingSum(accountTransactions3)).toEqual([])
+       })
+     })
+ 
+     // See the test fail
+       // FAIL  ./code-challenges.test.js
+       // fibFinder
+       //   ✓ takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence (1 ms)
+       // toGreatest
+       //   ✓ takes in an object and returns an array of the values sorted from least to greatest (1 ms)
+       // accumulatingSum
+       //   ✕ takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+ 
+       // ● accumulatingSum › takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+ 
+       //   ReferenceError: accumulatingSum is not defined
+ 
 // b) Create the function that makes the test pass.
+ 
+ // GREEN
+     //PSEUDOCODE:
+       // declare a function called "accumulatingSum"
+       // input: an array
+       // output: returns an array of the accumulating sum
+       // Process:
+       // Create a function called "accumulatingSum"
+       // In the beginning my method was return an array but adding the each element together by calling on their indexes: [array[0] + array[0 - 1] + array[0 - 2] + array[0- 3] after running my code I realized that the output was the sum of the total array and that is not was the goal is.
+       // After rethinking my process I then defined each variable since both given arrays have 4 elements. The output gave me my desired output for the first two arrays when I noticed I was hardcoding and I needed my function to be dynamic in order to work for all possible arrays and especially the three given arrays.
```

Comment:
This is awesome to take note of because it shows your ability to be self aware in order to draw yourself back in, rethink, and then go again.